spec/unit/dry/configurable/dsl_spec.rb

```diff
@@ -43,6 +43,15 @@
   end
 
   it 'compiles a setting with a constructor' do
+    setting = dsl.setting(:dsn, 'sqlite', constructor: ->(value) { "jdbc:#{value}" })
+
+    expect(setting.name).to be(:dsn)
+    expect(setting.value).to eql('jdbc:sqlite')
+  end
+
+  it 'supports but deprecates giving a constructor as a block' do
+    expect(Dry::Core::Deprecations).to receive(:announce)
```

Comment:
You can just [set a logger](https://github.com/dry-rb/dry-core/blob/e56a39776b54cd27e89431afb810b9c876d4fbfd/lib/dry/core/deprecations.rb#L75-L81) that would write to `log/deprecations.log` and this way you could verify if the messages look good. It's an approach we've been using so far.

Reply:
Please, @solnic, take a look at https://github.com/dry-rb/dry-configurable/pull/110/commits/4e24d6812708bdc43d6b50289783201b11ce6fb4. Do you mean like that? I used `StringIO` instead of an actual file to not have to mess with the filesystem. If that's what you meant, I'll squash both commits together.